pygac/calibration.py

```diff
@@ -536,30 +297,74 @@ def calibrate_thermal(counts, prt, ict, space, line_numbers, channel, spacecraft
-    space_convolved[-(wlength - 1) // 2:] = \
-        space_convolved[-((wlength + 1) // 2)]
+    space_convolved[-(wlength - 1) // 2:] = space_convolved[-((wlength + 1) // 2)]
 
     new_tprt = np.transpose(np.tile(tprt_convolved, (columns, 1)))
     new_ict = np.transpose(np.tile(ict_convolved, (columns, 1)))
     new_space = np.transpose(np.tile(space_convolved, (columns, 1)))
 
+    # Step 2. The radiance NBB sensed in each thermal AVHRR channel from the internal blackbody
+    # at temperature TBB is the weighted mean of the Planck function over the spectral response of the
+    # channel. [...]. Each thermal channel has one equation, which uses a centroid wavenumber labmda_c and an
+    # "effective" blackbody temperature TBB*. The two steps are:
+    # TsBB = A + B*TBB    (7.1.2.4-3)
+    # NBB = c1*nu_e^3/(exp(c2*nu_e/TsBB) - 1)    (7.1.2.4-3)
+    # where c1 = 1.1910427e-5 mW/m^2/sr/cm^{-4}, c2 = 1.4387752 cm K
+
     # calibrating thermal channel
 
     tBB = new_tprt
     tsBB = cal.a[chan] + cal.b[chan] * tBB
     nBB_num = (1.1910427 * 0.000010) * cal.c_wn[chan] ** 3
     nBB = nBB_num / (np.exp((1.4387752 * cal.c_wn[chan]) / tsBB) - 1.0)
 
+    # Step 3. Output from the two in-orbit calibration targets is used to compute a linear estimate of
+    # the Earth scene radiance NE. Each scanline, the AVHRR views the internal blackbody target and
+    # outputs 10 count values for each of the three thermal channel detectors; these are found in words
+    # 23 to 52 in the HRPT data stream. When the AVHRR views cold space, 10 counts from each of
+    # the five channel sensors are output and placed into words 52 to 102. (Table 4.1.3-1 describes
+    # how these data are multiplexed.) Count values for each channel are averaged together to smooth
+    # our random noise; often counts from five consecutive scanlines are averaged because it takes five
+    # lines to obtain a set of all four PRT measurements. The average blackbody count CBB and the
+    # average space count CS, together with blackbody radiance NBB and space radiance NS, explained
+    # in the next paragraph, are used to compute the linear radiance estimate NLIN,
+    # NLIN = NS + (NBB - NS)*(CS - CE)/(CS -CBB)    (7.1.2.4-5)
+    # where CE is the AVHRR count output when it views one of the 2,048 Earth targets.
+    # The Mercury-Cadmium-Telluride detectors used for channels 4 and 5 have a nonlinear response
+    # to incoming radiance. Pre-launch laboratory measurements show that:
+    #     a. scene radiance is a slightly nonlinear (quadratic) function of AVHRR output count,
+    #     b. the nonlinearity depends on the AVHRR operating temperature.
+    # It is assumed that the nonlinear response will persist in orbit. For the NOAA KLM series of
+    # satellites, NESDIS uses a radiance-based nonlinear correction method. In this method, the linear
+    # radiance estimate is first computed using a non-zero radiance of space, the NS term in Equation
```

Comment:
_I would add here that Ch3b radiance is computed with a linear approximation and that this is taken into account by the zero values in cal.n_s for this channel._